_includes/articles/payment-batching.md

```diff
@@ -0,0 +1,256 @@
+you'll be able to fee bump your transaction using either
+Child-Pays-for-Parent (CPFP) fee bumping or Replace-by-Fee (RBF) fee
+bumping.  In addition, the more unconfirmed children a transaction has,
+the more RBF fee bumping will cost as you'll have to pay for both the
+increased feerate of your transaction as well as for all the potential
+fees lost to miners when they remove any child transactions in order
+to accept your replacement.
+
+Note that these problems are not unique to batched payments---independent
+payments can have the same problem.  However, if an independent payment
+can't be fee bumped because the independent receiver spent their output,
+only that user is affected.  But if a single receiver of a batched
+payment spends their output to the point where fee bumping becomes
+impossible, all the other receivers of that transaction are also affected.
+
+As of Bitcoin Core 0.18 (April 2019), the limits are[^package-limits] that a
+group of related unconfirmed transactions may not exceed 101,000 vbytes
+in size, have more than 25 unconfirmed ancestors, or have more than 25
+descendants.  This size limit restricts batches to a maximum size of
+about 3,000 outputs and the descendant limit is easily reached if just a
+tiny percentage of those receiving a large batch respend their confirmed
+outputs.  It's also easy for any of the receivers to deliberately create
+transactions that reach one of these limits and prevent fee bumping if
+they know that you're relying on that capability.
+
+## Implementation
+
+Payment batching is extremely easy using certain existing wallet
+implementations, such as using Bitcoin Core's [sendmany][] RPC.  Check your software
+documentation for a function that allows you to send multiple payments.
+
+```bash
+bitcoin-cli sendmany "" '{
+  "bc1q5c2d2ue7x38hcw2ugk5q7y4ae7nw4r6vxcptu8": 0.1,
+  "bc1qztjzd7hpf2xmngr7zkgkxsvdqcv2jpyfgwgtsv": 0.2,
+  "bc1qsul9emtnz0kks939egx2ssa6xnjpsvgwq9chrw": 0.3
+}'
+```
+
+<!-- for max standard tx size: src/policy/policy.h:static const unsigned int MAX_STANDARD_TX_WEIGHT = 400000; -->
+
+If using your own implementation, you are probably already creating
+transactions with two outputs in most cases (a payment output and a
+change output), so it should be easy to add
+support for additional outputs.  The only notable consideration is that
+Bitcoin Core nodes (and most other nodes) will refuse to accept or relay
+transactions over 100,000 vbytes, so you should not attempt to send
+batched payments larger than this.
+
+## Recommendations summary
+
+1. Try to create systems where your users and customers don't expect
+   their payments immediately but are willing to wait for some time
+   (the longer the better).
+
+2. Use low-feerate consolidations to keep some large inputs available
+   for spending.
+
+3. Within each time window, send all payments together in the same
+   transaction.  For example, create an hourly [cronjob][] that sends all pending payments.
```

Comment:
Hourly seems too long for a "highfrequency spender", but then "highfrequency spender" is not further defined. I think I've only known one service to use one hour batches. Five minutes seemed the most common, but I've seen 20 second to 15 minutes be used successfully.

Reply:
I chose hourly because anyone waiting for six confirmations will occasionally have to wait considerably longer than an hour anyway, so it seems like a reasonable value and demonstrates even not-that-high-frequency spenders can use this technique effectively.

Reply:
Mh, yeah, I guess two payments per hour would be fairly high frequency compared to the average Bitcoin user. My frame of context is maybe a bit skewed here. ;)

Reply:
I think the 'for example' here covers us. If a service has a higher frequency of payments, they can change the cronjob frequency accordingly.